src/components/externalLinks/ExternalLinksCRUD/AddExternalLink.tsx

```diff
@@ -0,0 +1,321 @@
+                const areAllOptionsSelected = _selectedOption.findIndex((option) => option.value === '*') !== -1
+                const areAllOptionsAlredySeleted =
+                    Array.isArray(linksData[key].clusters) && linksData[key].clusters[0]?.value === '*'
+
+                if (areAllOptionsSelected && !areAllOptionsAlredySeleted) {
+                    linksData[key].clusters = [{ label: 'All clusters', value: '*' }]
+                } else if (!areAllOptionsSelected && areAllOptionsAlredySeleted) {
+                    linksData[key].clusters = []
+                } else if (areAllOptionsSelected && _selectedOption.length !== clusters.length) {
+                    linksData[key].clusters = _selectedOption.filter((option) => option.value !== '*')
+                } else if (!areAllOptionsSelected && _selectedOption.length === clusters.length - 1) {
+                    linksData[key].clusters = [{ label: 'All clusters', value: '*' }]
+                } else {
+                    linksData[key].clusters = _selectedOption
+                }
+                break
+            case 'onNameChange':
+                linksData[key].name = value as string
+                break
+            case 'onUrlTemplateChange':
+                linksData[key].urlTemplate = value as string
+                break
+            default:
+                break
+        }
+
+        setLinksData([...linksData])
+    }
+
+    const linksLen = linksData.length
+
+    const renderConfigureLinkActionColumn = (): JSX.Element => {
+        return (
+            <div className="configure-link-action-container">
+                {!selectedLink && (
+                    <div
+                        className="link-add-another fs-13 fw-6 mb-16 cursor"
+                        onClick={() => handleLinksDataActions('add')}
+                    >
+                        <AddIcon className="icon-dim-20 mr-8" /> Add another
+                    </div>
+                )}
+                {linksData &&
+                    linksData.map((link, idx) => {
+                        return (
+                            <Fragment key={`ConfigureLinkAction-${idx}`}>
+                                <ConfigureLinkAction
+                                    index={idx}
+                                    link={link}
+                                    clusters={clusters}
+                                    selectedClusters={
+                                        Array.isArray(link.clusters) && link.clusters[0]?.value === '*'
+                                            ? clusters
+                                            : link.clusters
+                                    }
+                                    monitoringTools={monitoringTools}
+                                    onMonitoringToolSelection={(key, selected) => {
+                                        handleLinksDataActions('onMonitoringToolSelection', key, selected)
```

Comment:
we should extract this as separate function

Reply:
1 level extraction done